app/Dockerfile

```diff
@@ -33,5 +54,64 @@ ENV HOST=0.0.0.0
 # Install application dependencies.
 # https://python-poetry.org/docs/basic-usage/#installing-dependencies
 RUN poetry install
+
 # Run the application.
 CMD ["poetry", "run", "python", "-m", "api"]
+
+#-------------------------------------------------------------------------------
+# Non-development build environment
+#-------------------------------------------------------------------------------
+FROM build as app-build
+
+ARG RUN_USER
+USER ${RUN_USER}
+
+# Set the application working directory.
+WORKDIR /app
+
+# install dependencies
+# use a project-local virtualenv to be easy to find
+ENV POETRY_VIRTUALENVS_IN_PROJECT true
+COPY pyproject.toml poetry.lock ./
+RUN poetry install --no-root --with app --without dev
+
+COPY . ./
+
+# Install application dependencies
+# The name of the wheel file comes from pyproject.toml
+RUN poetry build --format wheel && poetry run pip install 'dist/template_application_flask-0.1.0-py3-none-any.whl'
+
+#-------------------------------------------------------------------------------
+# Application only
```

Comment:
nit: the comment is a bit of a misnomer since the entire Dockerfile is for the application. Could we say something like "production image" or something

Reply:
Changed in next commit